rumqttc/src/lib.rs

```diff
@@ -434,6 +444,20 @@ impl MqttOptions {
         self.key_type
     }
 
+    /// Inject a rustls ClientConfig to use for making a TLS connection.
+    ///
+    /// Doing so implicitly enabled TLS. You must not set both this and the other TLS options
+    /// (set_ca, set_client_auth, set_alpn)
+    pub fn set_tls_client_config(&mut self, tls_client_config: Arc<ClientConfig>) -> &mut Self {
```

Comment:
The docs of `ClientConfig` say "Making one of these can be expensive, and should be once per process rather than once per connection.". It seems reasonable that someone might want to share the same ClientConfig between several different connections, perhaps some MQTT connections and some over other protocols, ideally without cloning it as it could take a decent chunk of memory. `Arc` seems to be the way to achieve this.